src/reader.rs

```diff
@@ -16,6 +16,11 @@ use memchr;
 
 #[derive(Clone)]
 enum TagState {
+    /// Initial state in which `StartText`, `Decl`, `Comment` or `Start` event
+    /// is expected
+    Init,
+    /// State after seeing the `<` symbol. After this state reader moves to
+    /// `Closed` state.
```

Comment:
It would be nice if all the variants were documented

Reply:
Yeah, I'm thinking about it but wanted to push earlier to give you time to review the code part. Will add